pkg/cli/idle/idle.go

```diff
@@ -653,42 +705,13 @@ func (o *IdleOptions) RunIdle() error {
 				continue
 			}
 
-			metadata, err := meta.Accessor(info.obj)
-			if err != nil {
-				fmt.Fprintf(o.ErrOut, "error: unable to mark service %q as idled: %v", serviceName.String(), err)
-				hadError = true
-				continue
-			}
-
-			gvks, _, err := scheme.Scheme.ObjectKinds(info.obj)
-			if err != nil {
-				fmt.Fprintf(o.ErrOut, "error: unable to mark service %q as idled: %v", serviceName.String(), err)
-				hadError = true
-				continue
-			}
-			// we need a versioned obj to properly marshal to JSON, so that we can compute the patch
-			mapping, err := o.Mapper.RESTMapping(gvks[0].GroupKind(), gvks[0].Version)
-			if err != nil {
-				fmt.Fprintf(o.ErrOut, "error: unable to mark service %q as idled: %v", serviceName.String(), err)
-				hadError = true
-				continue
-			}
-
-			oldData, err := json.Marshal(info.obj)
-			if err != nil {
+			if err := patchObjWithIdleAnnotations(info.service, refsWithScale, nowTime); err != nil {
 				fmt.Fprintf(o.ErrOut, "error: unable to mark service %q as idled: %v", serviceName.String(), err)
 				hadError = true
 				continue
```

Comment:
do we want to continue here or do we try to patch the endpoints despite the error?

Reply:
Good question; I followed the existing pattern here. Even if this fails, and endpoints succeeds, it seems we can also bail (and continue) if the scalable resource updates fail in any way. There's no atomicity across the changes we're trying to make.